src/features/todosState.ts

```diff
@@ -1,15 +1,84 @@
-import { createAsyncThunk, createSelector, createSlice } from '@reduxjs/toolkit'
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
 
 import type { RootState } from '../app/store'
 import { todosController } from '../networking/controllers/todos'
-import { Todo } from '../types/globalTypes'
+import { NewItem, Todo } from '../types/globalTypes'
 
 export interface TodosState {
   todos: Todo[]
   loading: boolean
   error: string | undefined
 }
 
+export interface ItemCheck {
+  id: string
+  title: string
+  description: string
+  toCheck: boolean
+}
+
+export const clearTodoAPI = createAsyncThunk<
+  void,
+  string,
+  { rejectValue: string }
+>('todos/deleteTodosAPI', async (id: string, thunkAPI) => {
+  try {
+    return await todosController.deleteTodo(id)
```

Comment:
Well done 💪 
I leave you a good article about `return vs return await`. Head-up: you did great.

https://jakearchibald.com/2017/await-vs-return-vs-return-await/